pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -1281,13 +1281,46 @@ func (self *AWSCloud) TCPLoadBalancerExists(name, region string) (bool, error) {
 	return false, nil
 }
 
+// Retrieves instance's vpc id from metadata
+func (self *AWSCloud) findVPCId() (string, error) {
+
+	metadata := self.awsServices.Metadata()
+	macs_bytes, err := metadata.GetMetaData("network/interfaces/macs/")
+	if err != nil {
+		err = fmt.Errorf("Could not list interfaces of the instance", err)
+		return "", err
+	}
+
+	// loop over interfaces, first vpc id returned wins
+	for _, mac_path := range strings.Split(string(macs_bytes), "\n") {
+		if len(mac_path) == 0 {
+			continue
+		}
+		url := fmt.Sprintf("network/interfaces/macs/%svpc-id", mac_path)
+		vpc_id_bytes, _ := metadata.GetMetaData(url)
+		if err != nil {
+			continue
+		}
+		return string(vpc_id_bytes), nil
+	}
+	err = fmt.Errorf("Could not find VPC id in instance metadata")
+	return "", err
+}
+
 // Find the kubernetes VPC
 func (self *AWSCloud) findVPC() (*ec2.VPC, error) {
 	request := &ec2.DescribeVPCsInput{}
 
-	name := "kubernetes-vpc"
-	filters := []*ec2.Filter{newEc2Filter("tag:Name", name)}
-	request.Filters = self.addFilters(filters)
+	// find by vpc_id from metadata, else use hardcoded vpc name
```

Comment:
I think it is perfectly reasonable to require the node to be running in the VPC, and so I think we can just remove the hard-coded fallback.